### PR TITLE
removing extra bool array as it is not needed

### PR DESCRIPTION
**Pull request by armenpoghosov**
removing extra bool array as it is not needed for the search of reachable input nodes

### CODE REVIEW

File: tensorflow/cc/framework/gradients.cc [hunk 1]
```diff
@@ -167,11 +167,12 @@ Status SymbolicGradientBuilder::BackpropAlongEdge(const Output& dst_grad,
 std::vector<bool> SymbolicGradientBuilder::GetReachableNodes() {
   std::vector<bool> reachable_nodes(scope_.graph()->num_node_ids(), false);
   std::deque<Node*> queue;
-  std::vector<bool> visited(scope_.graph()->num_node_ids(), false);
   for (const Output& out : outputs_) {
-    if (!reachable_nodes[out.node()->id()]) {
-      queue.push_back(out.node());
-      reachable_nodes[out.node()->id()] = true;
+    Node* const out_node = out.node();
+    const int out_node_id = out_node->id();
```
Comment (skye) [inline tensorflow/cc/framework/gradients.cc]: nit: I think creating these out variables doesn't increase readability and makes the code longer. Can you change it back to using the inline function calls?

File: tensorflow/cc/framework/gradients.cc [hunk 2]
```diff
@@ -180,10 +181,12 @@ std::vector<bool> SymbolicGradientBuilder::GetReachableNodes() {
     queue.pop_front();
     for (const Edge* e : n->in_edges()) {
       if (e->IsControlEdge()) continue;
-      if (visited[e->src()->id()]) continue;
-      queue.push_back(e->src());
-      reachable_nodes[e->src()->id()] = true;
-      visited[e->src()->id()] = true;
+      Node* const src_node = e->src();
+      const int src_node_id = src_node->id();
```
Comment (skye) [inline tensorflow/cc/framework/gradients.cc]: Same here

**Review by skye — CHANGES_REQUESTED** [review]
Looks good except for the minor style nits, thanks!

File: tensorflow/cc/framework/gradients.cc (hunk 1 above)
Comment (armenpoghosov) [reply tensorflow/cc/framework/gradients.cc]: done

File: tensorflow/cc/framework/gradients.cc (hunk 2 above)
Comment (armenpoghosov) [reply tensorflow/cc/framework/gradients.cc]: done

**Review by skye — APPROVED** [review]

**Pull request merged**